### DISCUSSION TITLE
About one Payload repo and different frontend repo

### DISCUSSION THREAD

**Discussion by KBambharoliya** [payloadcms/payload#9513] [General]
My scenario here is i do have one payload repo that have all the collections added there but we are thinking to have different frontend repo that is going to be used in different frontend language some repo will be Reactjs project some of them NextJS project. we will use payload api there.
so my question is can i get payload.find or getPayloadHMR in frontend? I want to fetch data using payload.find in frontend can i use it ?

**Comment by ckruppe** [comment]
I don't think so. The getPayload function is an local api. So the frontend and payload have to live together. But thats why the rest/graphql api exists. You could fetch the api endpoint in your server components to achieve the same outcome as you do with payload.find.
Also you can use payloads lokal api only in the backend as these are direct database calls.

I can imagine if you have payload installed and the configs copied over to your frontends then it would work too. (As its "only" db calls). But that defeats the purpose a bit for having it extra.

So no you cant really but thats no payload restriction. Its more an technical one as your Browser cant call the DB directly. 

Use the Rest Api and copy over the types generated by your payload repo. Then you have type Safe api calls. :)

**Reply by spaaacetoast** [reply]
Maybe im not understanding correctly but there would be no difference in using payload local api in the shared project vs loading the config in another frontend, be it a meta framework like astro or solidstart? as long as its used server side, right (db calls)?

**Reply by KBambharoliya** [reply]
I tried to add payload.config.ts file in frontend but still i don't get payload local api in page.tsx file to fetch data.

**Reply by spaaacetoast** [reply]
> I tried to add payload.config.ts file in frontend but still i don't get payload local api in page.tsx file to fetch data.
> 

feel free to check out my repo https://github.com/spaaacetoast/shifter. on the astro homepage I use the payload local api server side and display the results in html as quick test

**Comment by yonnic** [comment]
i am using this rest client which is very close to the local API in my astro project
https://www.npmjs.com/package/payload-rest-client/

**Reply by KBambharoliya** [reply]
Thank  you @yonnic . I will use payload-rest-client.
